src/v3/src/components/Widget/index.tsx

```diff
@@ -82,6 +82,7 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
   });
   const [message, setMessage] = useState<IdxMessage | undefined>();
   const [idxTransaction, setIdxTransaction] = useState<IdxTransaction | undefined>();
+  const [previousTransaction, setPreviousTransaction] = useState<IdxTransaction | undefined>();
```

Comment:
you don't need this state, see line 88

Reply:
Oh yea! Great catch, I actually tried to use this initially, I think it may have something to do with when it gets updated but this previous transaction did not have the correct stateHandle. The stateHandle in this object matched that from the current transaction, which is why I created a new state object. 

When I saw that I didnt debug more into why, mostly because I didnt want to make any changes that would negatively impact the code that is using this ref.

Reply:
Here is what I was talking about: 
<img width="432" alt="image" src="https://user-images.githubusercontent.com/97472729/187300760-d0967cbe-d856-4f15-a2e0-5582d1f806c8.png">

Reply:
Ref is the existing ref that was created, current transaction is the current transaction, and prev is the new state object I created. 

My fear was changing something with that prev transaction ref and causing breakage for whatever was using it. Especially if this logic is going to go away after a bug fix, I didn't think it would be worth it to put it a bunch of effort into that.

Reply:
the last one in your log (probably what you tracked in state), does not look like a stateHandle, but more like a token instead, what you get if you decode it?

Reply:
This is what I get when I decode it, it is directly from the `stateHandle` property that is in the transaction context. 

It is 100% the difference of which transaction is being used. The ref uses the last successful transaction whereas this state object uses the "currentTransaction". Maybe the name is confusing. I think I should just create a stateProp called `savedStateHandle` or `stateHandle` or something like that to avoid confusion. 

<img width="1271" alt="image" src="https://user-images.githubusercontent.com/97472729/187302148-e8c82e82-14ba-495f-9a13-f23324f233fb.png">

Reply:
You can see here, this is the transaction we are using the stateHandle from for this workaround: https://github.com/okta/okta-signin-widget/pull/2752/files#diff-114b8aded3d172ffc719a9452df5b38cfe8f9e82362d922bc0ca4f7f4440005aR111

Reply:
Where as the ref uses the last successful transaction. So maybe the best idea is to just save the stateHandle as a state prop and not the entire transaction.

Reply:
ohh, this one is for the terminal state issue, authjs clears transaction related data, I don't think a workaround can be applied in widget side. I was suggesting https://github.com/okta/okta-signin-widget/blob/21b98876539d6c48a7108a44de337027085f41d7/src/v2/view-builder/utils/LinksUtil.js#L121

Reply:
Ahh, so the above solution does work, see the integration test, (albeit not the greatest). I took at look at that `getBackToSignInLink` but ultimately didnt do it because in this scenario, the cancel action exists on the transaction. How will we know when to use it? Or are you saying to always use this logic regardless if `cancel` action exists or not?

Reply:
As I said before, it's how v2 works, and authjs currently has issue for this scenario. So I would suggest to follow v2 instead of hacking another solution.

Reply:
Yep, I recall :D but this is how v2 handles it today: https://github.com/okta/okta-signin-widget/blob/21b98876539d6c48a7108a44de337027085f41d7/src/v2/view-builder/views/TerminalView.js#L158-L164

This only happens if the cancel object doesnt exist, in this scenario, the cancel action is present. So are you saying to always use this logic even if the cancel action is present?

Reply:
I am not against doing this, I just want to make sure this doesnt cause any other issues by not using the cancel action when it is expected to be used.

Reply:
v3 should work if v2 works, which does not send cancel request in the current version (based on what i read from code)

Reply:
V2 is broken in this scenario (and other scenarios like this) as well (same exact issue, it calls introspect instead of cancel) 😅

Reply:
The only fix would be to always use this `getBackToSignInLink` logic in v3 until a fix can be applied to auth-js. Or what I have above with the stateHandle. 

Which method is preferred?